Malmo/src/AgentHost.cpp

```diff
@@ -686,33 +645,28 @@ namespace malmo
 
                 // Add some diagnostics of our own before this gets to the agent:
                 if (this->video_server || this->luminance_server || this->depth_server || this->colourmap_server) {
-                    for (auto &vd : mission_ended->MissionDiagnostics().VideoData()) {
+                    for (auto &vd : mission_ended.videoDataAttributes()) {
                         boost::shared_ptr<VideoServer> vs = 0;
-                        if (vd.frameType() == "VIDEO")
+                        if (vd.frame_type == "VIDEO")
                             vs = this->video_server;
-                        else if (vd.frameType() == "DEPTH_MAP")
+                        else if (vd.frame_type == "DEPTH_MAP")
                             vs = this->depth_server;
-                        else if (vd.frameType() == "LUMINANCE")
+                        else if (vd.frame_type == "LUMINANCE")
                             vs = this->luminance_server;
-                        else if (vd.frameType() == "COLOUR_MAP")
+                        else if (vd.frame_type == "COLOUR_MAP")
                             vs = this->colourmap_server;
                         if (vs) {
-                            vd.framesReceived(vs->receivedFrames());
-                            vd.framesWritten(vs->writtenFrames());
+                            vd.frames_received =  vs->receivedFrames();
+                            vd.frames_written = vs->writtenFrames();
                         }
                     }
-                    std::ostringstream oss;
-                    xml_schema::namespace_infomap map;
-                    map[""].name = xml_namespace;
-                    map[""].schema = "MissionEnded.xsd";
-                    xml_schema::flags flags = xml_schema::flags::dont_initialize;
-                    malmo::schemas::MissionEnded_(oss, *mission_ended, map, "UTF-8", flags);
-                    xml.text = oss.str();
+
+                    xml.text = mission_ended.toXml();
                 }
             }
-            catch (const xml_schema::exception& e) {
+            catch (const std::exception& e) {
                 std::ostringstream oss;
-                oss << "Error parsing MissionEnded message XML: " << e.what() << " : " << e << ":" << xml.text.substr(0, 20) << "...";
+                oss << "Error processing MissionEnded message XML: " << e.what() << " : " << ":" << xml.text.substr(0, 20) << "...";
```

Comment:
Did you mean to miss out the << e ?

Reply:
Yes, std::exception does not have a cout << member. Still writes out the human readable message but could do more for debugging.

Reply:
Fair enough. You could get rid of the superfluous colon then!